R/agglomerate.R

```diff
@@ -155,16 +155,22 @@ setGeneric("mergeFeaturesByRank",
 #'
 #' @export
 setMethod("agglomerateByRank", signature = c(x = "SummarizedExperiment"),
-    function(x, rank = taxonomyRanks(x)[1], onRankOnly = FALSE, na.rm = FALSE,
+    function(x, rank = taxonomyRanks(x)[1], onRankOnly = TRUE, na.rm = FALSE,
         empty.fields = c(NA, "", " ", "\t", "-", "_"), ...){
         # input check
         if(nrow(x) == 0L){
             stop("No data available in `x` ('x' has nrow(x) == 0L.)",
                  call. = FALSE)
         }
-        if(!.is_non_empty_string(rank)){
-            stop("'rank' must be an non empty single character value.",
-                call. = FALSE)
+        if (!is.null(rank)) {
+            if (!.is_non_empty_string(rank)){
+                stop("'rank' must be a non-empty single character value.", call. = FALSE)
+            }
+            if (!(rank %in% taxonomyRanks(x))) {
+                stop("'rank' must be a value from 'taxonomyRanks()'", call. = FALSE)
+            }
+        } else {
+            rank <- taxonomyRanks(x)[1]
         }
```

Comment:
INsteqad of this, change default behavior of rank in agglomerateByPrevalence. --> there is no need to change the behavior here, NULL (i.e. no agglomeration) does nothing

However, you should change the default rank to NULL in agglomeratebyPrevalence --> when NULL, agglomerateByRank is skipped.

Reply:
Even though, we discussed in the issue otherwise.... sorry about that